### PR TITLE
Fixed issues mentioned in #882

### PR DESCRIPTION
**Pull request by thejohncrafter**
There was some issues with Schematics (see #882) : 
- `!=` check instead of `==` [(see here)](https://github.com/SpongePowered/SpongeCommon/pull/882#discussion_r77175692)
- Builder returning `null` instead of `this` [(see here)](https://github.com/SpongePowered/SpongeCommon/pull/882#discussion_r77203029)
- `NullPointerException` in builder [(see here)](https://github.com/SpongePowered/SpongeCommon/pull/882#discussion_r77206623)

I'm not completely sure about the last fix - did I do it the right way ?

### CODE REVIEW

**Comment by Aaron1011** [comment]
Pinging @Deamon5550 for review.

File: src/main/java/org/spongepowered/common/world/schematic/SpongeSchematicBuilder.java [hunk 1]
```diff
@@ -147,12 +148,18 @@ public Schematic build() throws IllegalArgumentException {
...
             this.view.getBlockWorker(SpongeImpl.getImplementationCause()).iterate((v, x, y, z) -> {
                 volume.setBlock(x, y, z, v.getBlock(x, y, z), SpongeImpl.getImplementationCause());
+                Optional<TileEntity> tile = v.getTileEntity(x, y, z);
+                if (tile.isPresent()) {
+                    tiles.put(new Vector3i(x, y, z), tile.get().createArchetype());
+                }
             });
             return new SpongeSchematic(volume, tiles, this.metadata);
+        } else {
+            return new SpongeSchematic((SpongeArchetypeVolume) this.volume, this.metadata);
```
Comment (Deamon5550) [inline src/main/java/org/spongepowered/common/world/schematic/SpongeSchematicBuilder.java]: don't nest this in an else its pointless

**Comment by Deamon5550** [comment]
Looks fine otherwise.

File: src/main/java/org/spongepowered/common/world/schematic/SpongeSchematicBuilder.java (hunk 1 above)
Comment (thejohncrafter) [inline src/main/java/org/spongepowered/common/world/schematic/SpongeSchematicBuilder.java]: Just for more readability... but ok :wink:

**Comment by Deamon5550** [comment]
Merged in 2beaed4

**Pull request closed without merging**